packages/fiori/src/UploadCollection.js

```diff
@@ -0,0 +1,354 @@
+	draggingFiles,
+} from "./upload-utils/BodyDragAndDrop.js";
+import UploadCollectionDnDOverlayMode from "./types/UploadCollectionDnDMode.js";
+
+// Template
+import UploadCollectionTemplate from "./generated/templates/UploadCollectionTemplate.lit.js";
+
+// Styles
+import UploadCollectionCss from "./generated/themes/UploadCollection.css.js";
+
+/**
+ * @public
+ */
+const metadata = {
+	tag: "ui5-upload-collection",
+	properties: /** @lends sap.ui.webcomponents.fiori.UploadCollection.prototype */ {
+		/**
+		 * Defines the mode of the <code>ui5-upload-collection</code>.
+		 * <br><br>
+		 * <b>Note:</b> Available options are <code>None</code>, <code>SingleSelect</code>,
+		 * <code>MultiSelect</code>, and <code>Delete</code>.
+		 *
+		 * @type {string}
+		 * @defaultvalue "None"
+		 * @public
+		 */
+		mode: {
+			type: ListMode,
+			defaultValue: ListMode.None,
+		},
+
+		/**
+		 * Allows you to set your own text for the 'No data' description.
+		 *
+		 * @type {string}
+		 * @public
+		 */
+		noDataDescription: {
+			type: String,
+		},
+
+		/**
+		 * Allows you to set your own text for the 'No data' text.
+		 *
+		 * @type {string}
+		 * @public
+		 */
+		noDataText: {
+			type: String,
+		},
+
+		/**
+		 * By default there will be drag and drop overlay shown over the <code>ui5-upload-collection</code> when files
+		 * are dragged. If you don't intend to use drag and drop, set this property to <code>true</code>
+		 * <br><br>
+		 * <b>Note:</b> It is up to the application developer to add handler for <code>drop</code> event and handle it.
+		 * <code>ui5-upload-collection</code> only shows an overlay.
+		 *
+		 * @type {boolean}
+		 * @public
+		 */
+		noDnd: {
+			type: Boolean,
+		},
+
+		/**
+		 * Indicates what overlay to show when files are being dragged.
+		 *
+		 * @private
+		 */
+		_dndOverlayMode: {
+			type: String,
+			defaultValue: UploadCollectionDnDOverlayMode.None,
+		},
+	},
+	managedSlots: true,
+	slots: /** @lends sap.ui.webcomponents.fiori.UploadCollection.prototype */ {
+		/**
+		 * Defines the items of the <code>ui5-upload-collection</code>.
+		 * <br><b>Note:</b> Use <code>ui5-upload-collection-item</code> for the intended design.
+		 *
+		 * @type {HTMLElement[]}
+		 * @slot
+		 * @public
+		 */
+		"default": {
+			propertyName: "items",
+			type: HTMLElement,
+		},
+
+		/**
+		 * Defines the <code>ui5-upload-collection</code> header.
+		 *
+		 * @type {HTMLElement[]}
+		 * @slot
+		 * @public
+		 */
+		header: {
+			type: HTMLElement,
+		},
+	},
+	events: /** @lends sap.ui.webcomponents.fiori.UploadCollection.prototype */ {
+		/**
+		 * Fired when the Delete button of any item is pressed.
+		 * <br><br>
+		 * <b>Note:</b> A Delete button is displayed on each item,
+		 * when the <code>ui5-upload-collection</code> <code>mode</code> property is set to <code>Delete</code>.
+		 * @event
+		 * @param {HTMLElement} item The <code>ui5-upload-collection-item</code> which was renamed.
+		 * @public
+		 */
+		fileDeleted: {
```

Comment:
Didn't we agree that all such events go to item level? Or maybe I'm wrong

Reply:
Whether files have delete button or not is decided by the "mode" property of UploadCollection. So the items don't know of it.